### DISCUSSION TITLE
Moved `lang` directory in Laravel 9

### DISCUSSION THREAD

**Discussion by andrey-helldar** [Laravel-Lang/common#31] [Q&A]
See: https://github.com/laravel/laravel/commit/7156a27cdbaa0e69bfd04bb5ef083d9906bf543c

And there: https://github.com/laravel/framework/commit/e6c8aaea886d35cc55bd3469f1a95ad56d53e474#diff-0a80606049a107748776bea5151a61fbec2afefe3361c340a5de9c8c1ccdf9c8R322-R329

**resources/lang/en → lang/en**

https://github.com/laravel/laravel/tree/master/lang

Hm... Laravel 8: https://github.com/laravel/framework/blob/8.x/src/Illuminate/Foundation/Application.php#L415-L426

**Comment by mohamedsabil83** [comment]
seems they finally decide to move it to the root. As I remembered, they started thinking about doing that since Laravel 7.

**Reply by andrey-helldar** [reply]
Yes, I [found](https://github.com/laravel/framework/blob/8.x/src/Illuminate/Foundation/Application.php#L415-L426) the same code in Laravel 8, but by default the folder was still in `resources` directory.

At first I wrote to keep up to date, but then I saw the code in Laravel 8 and did not delete this discussion 🙂

At least with the release of Laravel 9, many will say something like "I'm watching a video tutorial, it has a `resources/lang` folder, but I don't. What's the problem?" or "I am saving files to the lang folder, but the application does not see it (the `resources/lang` folder exists)".

Here is a search engine, perhaps, and will lead them to this page 😁

**Reply by mohamedsabil83** [reply]
Yep, it's a good idea 😎

**Reply by luisprmat** [reply]
This issue should not affect the `lang` package, you just have to adjust the `publisher` package so that it checks where it should copy the translations if in `/lang` or `/resources/lang`.

**Reply by andrey-helldar** [reply]
Yes, it won't.

I left a description here just so that when they start looking for the problem after the upgrade, they will find our package :)

So to speak, a small attempt at promotion 😁
